client/state/data-layer/wpcom/read/streams/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import { find, identity } from 'lodash';
+import { translate } from 'i18n-calypso';
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import warn from 'lib/warn';
+import {
+	requestInflight,
+	markRequestInflight,
+	completeRequest
+} from 'lib/inflight';
+import {
+	READER_STREAMS_PAGE_REQUEST
+} from 'state/action-types';
+import { receivePage } from 'state/reader/streams/actions';
+import { errorNotice } from 'state/notices/actions';
+
+const streamToPathMatchers = [
+	// [ regex, version, path, advice ]
+	// ordering here is by how often we expect each stream type to be used
+	// search is linear, so putting common things near the front can be helpful
+	[ /^following$/, 'v1.2', '/read/following' ],
+	[ /^search:/, 'v1.2', '/read/search' ],
+	[ /^feed:/, 'v1.2', '/read/feed/:feed/posts' ],
+	[ /^site:/, 'v1.2', '/read/sites/:site/posts' ],
+	[ /^featured:/, 'v1.2', '/read/sites/:site/featured' ],
+	[ /^a8c$/, 'v1.2', '/read/a8c' ],
+	[ /^likes$/, 'v1.2', '/read/liked' ],
+	[ /^recommendations_posts$/, 'v1.2', '/read/recommendations/posts' ],
+	[ /^custom_recs:/, 'v1.2', '/read/recommendations/posts' ],
+	[ /^tag:/, 'v1.2', '/read/tags/:tag/posts' ],
+	[ /^list:/, 'v1.2', '/read/list/:owner/:slug/posts' ],
+	[ /^featured:/, 'v1.2', '' ]
+];
+
+function apiForStream( streamId ) {
+	return find( streamToPathMatchers, ( matcher ) => matcher[ 0 ].test( streamId ) );
+}
+
+export function keyForRequest( action ) {
+	const { streamId, query } = action;
+	const actionString = !! query
+		? Object.keys( query )
```

Comment:
can probably leave out the ternary operator by instead making it:
`Object.keys( query || {}  )`